symbology/read-symbols-from-mobile-style-file/src/main/java/com/esri/samples/read_symbols_from_mobile_style_file/ReadSymbolsFromMobileStyleFileController.java

```diff
@@ -0,0 +1,356 @@
+   */
+  private void updateSymbolPreview(MultilayerPointSymbol multilayerPointSymbol) {
+    // remove the previously displayed image view
+    symbolPreviewHBox.getChildren().clear();
+
+    // create an image and image view from the symbol
+    ListenableFuture<Image> symbolImageFuture = multilayerPointSymbol.createSwatchAsync(0x00000000, 1);
+    try {
+      Image symbolImage = symbolImageFuture.get();
+      ImageView symbolImageView = new ImageView(symbolImage);
+      // display the image view in the preview area
+      symbolPreviewHBox.getChildren().add(symbolImageView);
+
+    } catch (InterruptedException | ExecutionException e) {
+      new Alert(Alert.AlertType.ERROR, "Error creating preview ImageView from provided MultilayerPointSymbol" + e.getMessage()).show();
+    }
+  }
+
+  /**
+   * Updates the size of the symbol preview and the label when the slider is interacted with.
+   */
+  @FXML
+  private void updateSymbolSize() {
+    // get the slider value and convert to a string
+    String sizeValue = Integer.toString((int) sizeSlider.getValue());
+    // display the value in the label
+    sizeLabel.setText(sizeValue + " px");
+
+    // update the preview
+    buildCompositeSymbol();
+  }
+
+  /**
+   * Adds graphics to the map view on mouse clicks.
+   * @param e mouse button click event
+   */
+  @FXML
+  private void handleMouseClicked(MouseEvent e){
+    if (e.getButton() == MouseButton.PRIMARY && e.isStillSincePress()) {
+      // convert clicked point to a map point
+      Point mapPoint = mapView.screenToLocation(new Point2D(e.getX(), e.getY()));
+
+      // create a new graphic with the point and symbol
+      Graphic graphic = new Graphic(mapPoint, faceSymbol);
+      graphicsOverlay.getGraphics().add(graphic);
+    }
+  }
+
+  /**
+   * Clears all the graphics from the graphics overlay.
+   */
+  @FXML
+  private void resetView() {
+    graphicsOverlay.getGraphics().clear();
+  }
+
+  /**
+   * Stops and releases all resources used in application.
+   */
+  public void terminate() {
+
+    if (mapView != null) {
+      mapView.dispose();
+    }
+  }
+
+  /**
+   * Shows the available symbol of the SymbolStyleSearchResult in the symbol selection list view.
+   */
+  private class SymbolLayerInfoListCell extends ListCell<SymbolStyleSearchResult> {
+
+    private SymbolLayerInfoListCell() {
+      // set the cell to display only a graphic
+      setContentDisplay(ContentDisplay.GRAPHIC_ONLY);
+    }
+
+    @Override
+    protected void updateItem(SymbolStyleSearchResult item, boolean empty) {
+      super.updateItem(item, empty);
+
+      if (item == null || empty) {
+        // if the item in the list view is an empty item, show nothing
+        setGraphic(null);
+      } else {
+        // get the symbol from the list view entry, and create an image from it
+        ListenableFuture<Image> symbolImageFuture = item.getSymbol().createSwatchAsync(0x00000000, 1);
```

Comment:
I think you'd want a `doneListener` here, otherwise there's no point in doing this asynchronously.

Reply:
I have added the `doneListener` but I think it adds a slightly weird UI bug (see gif).
A symbol appears under `eyes`, and when selecting a different one it disappears and leaves the 'empty' box, what it should have been before.
![3C2AAE7E-84BF-43BD-A389-C07B93A29F76](https://user-images.githubusercontent.com/44193148/61698455-0f1be580-ad31-11e9-86e4-aa83555e7fc5.GIF)

Reply:
You don't want to set the cell's graphic asynchronously. Try setting it to an empty `ImageView`, then update the view's image when the swatch is ready.

```java
ImageView symbolImageView = new ImageView();
setGraphic(symbolImageView);
ListenableFuture<Image> symbolImageFuture = item.getSymbol().createSwatchAsync(0x00000000, 1);
symbolImageFuture.addDoneListener(() -> {
    try {
          // get the symbol swatch
          Image symbolImage = symbolImageFuture.get();
          // update the image view with the symbol swatch
          symbolImageView.setImage(symbolImage);
    } catch (InterruptedException | ExecutionException e) {
          new Alert(Alert.AlertType.ERROR, "Error creating preview image for symbol in mobile style file" + e.getMessage()).show();
    }
});
```

Reply:
Thanks for that! I see how it fixes it, but obviously didn't think about this causing the issue to begin with.